### PR TITLE
port geometry wrappers from sbndcode

### PR DESCRIPTION
**Pull request by chilge**
port T. Brooks's GeoWrapper tool from `sbndcode` in order to make `icaruscode` play nicely with `sbnci`

### CODE REVIEW

**Comment by SFBayLaser** [comment]
I have made @PetrilloAtWork a reviewer...

File: fcl/gen/genie/ci_simulation_genie_bnb_icarus.fcl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# file: ci_simulation_genie_bnb_icarus.fcl
+# author: chilgenb@fnal.gov
+# created: Nov 1 2021
+# description: wrapper around simulation_genie_icarus_bnb.fcl
+#              that enables IFDH access of the GENIE flux
+#              files instead of DIRECT, required to run on grid
+
+#include "simulation_genie_icarus_bnb.fcl"
+
+physics.producers.generator: @local::icarus_genie_BNB
```
Comment (PetrilloAtWork) [inline fcl/gen/genie/ci_simulation_genie_bnb_icarus.fcl:10]: Isn't this line redundant?
```suggestion
```
I would expect this to be already set by `simulation_genie_icarus_bnb.fcl`.

File: fcl/gen/genie/ci_simulation_genie_nue_icarus.fcl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# file: ci_simulation_genie_nue_icarus.fcl
+# author: chilgenb@fnal.gov
+# created: Nov 1 2021
+# description: wrapper around simulation_genie_icarus_Mar2019_nue.fcl
+#              that enables IFDH access of the GENIE flux
+#              files instead of DIRECT, required to run on grid
+
+#include "simulation_genie_icarus_Mar2019_nue.fcl"
+
+physics.producers.generator: @local::icarus_genie_BNB
```
Comment (PetrilloAtWork) [inline fcl/gen/genie/ci_simulation_genie_nue_icarus.fcl:10]: Ditto.
```suggestion
```

**Review by PetrilloAtWork — COMMENTED** [review]
Unrequested review: please check that the settings of `generator` in the included FHiCL files are not fully overwritten by the new ones.

File: fcl/gen/genie/ci_simulation_genie_bnb_icarus.fcl (hunk 1 above)
Comment (chilge) [reply fcl/gen/genie/ci_simulation_genie_bnb_icarus.fcl]: Indeed it is!

File: fcl/gen/genie/ci_simulation_genie_nue_icarus.fcl (hunk 2 above)
Comment (chilge) [reply fcl/gen/genie/ci_simulation_genie_nue_icarus.fcl]: yup

**Comment by PetrilloAtWork** [comment]
I did not realize that I _was_ requested a review.
I have a preliminary question: why is this not moved to `sbncode`?

**Comment by chilge** [comment]
@PetrilloAtWork good question. This `TPCGeoAlg` class uses the geometry service. My understanding is that, since `sbncode` does not depend on the experiment code, this precludes such a class from living there. At some point someone (maybe me?) should abstract out the common bits to live in `sbncode` with very light implementations in the experiment areas.

**Comment by PetrilloAtWork** [comment]
> This `TPCGeoAlg` class uses the geometry service. My understanding is that, since `sbncode` does not depend on the experiment code, this precludes such a class from living there. At some point someone (maybe me?) should abstract out the common bits to live in `sbncode` with very light implementations in the experiment areas.

To comment more precisely, I will need to see the details of the code, but that never stopped me anyway.
Assuming that you refer to `geo::Geometry` (i.e. LArSoft `Geometry` service):
1. a class using LArSoft `Geometry` service _can_ live in any repository that depends on `larcore`, including `sbncode`;
2. a job using ICARUS geometry (i.e. a FHiCL job configuration where `Geometry` service is set to ICARUS geometry) should live (and can run) only in repositories depending on `icaruscode`.

**Comment by chilge** [comment]
Ahh I see. Thanks for clarifying that. I believe this class matches case 1 in your list. I will put the `TPCGeoAlg` class in `sbncode`. Thank you @PetrilloAtWork .

**Pull request closed without merging**